From 98749a24de6b0f3b1946206638b4dc01755d5e79 Mon Sep 17 00:00:00 2001
From: William Wu <william.wu@rock-chips.com>
Date: Mon, 17 Sep 2018 19:27:45 +0800
Subject: [PATCH] usb: dwc3: rockchip: fix build error if !CONFIG_USB

This patch fix the following build error if CONFIG_USB
is disabled and only support dwc3 gadget mode:

dwc3-rockchip.c:894: undefined reference to `usb_add_hcd'
dwc3-rockchip.c:622: undefined reference to `usb_remove_hcd'
......
dwc3-rockchip-inno.c:268: undefined reference to `usb_remove_hcd'
dwc3-rockchip-inno.c:286: undefined reference to `usb_add_hcd'

Change-Id: Iaa51ccc642abf5741fcd0d918967954c840240d5
Signed-off-by: William Wu <william.wu@rock-chips.com>
---
 drivers/usb/dwc3/Kconfig | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/dwc3/Kconfig b/drivers/usb/dwc3/Kconfig
index 3fbf82d494f9..8b444dc0a50f 100644
--- a/drivers/usb/dwc3/Kconfig
+++ b/drivers/usb/dwc3/Kconfig
@@ -98,8 +98,8 @@ config USB_DWC3_OF_SIMPLE
 
 config USB_DWC3_ROCKCHIP
 	tristate "Rockchip Platforms"
-	depends on EXTCON && (ARCH_ROCKCHIP || COMPILE_TEST)
-	depends on OF
+	depends on OF && EXTCON && (ARCH_ROCKCHIP || COMPILE_TEST)
+	depends on USB=y || USB=USB_DWC3
 	default USB_DWC3
 	help
 	  Support of USB2/3 functionality in Rockchip platforms.
@@ -108,6 +108,7 @@ config USB_DWC3_ROCKCHIP
 config USB_DWC3_ROCKCHIP_INNO
 	tristate "Rockchip Platforms with INNO PHY"
 	depends on OF && COMMON_CLK && ARCH_ROCKCHIP
+	depends on USB=y || USB=USB_DWC3
 	default USB_DWC3
 	help
 	  Support of USB2/3 functionality in Rockchip platforms
-- 
2.35.3

